test/e2e_node/checkpoint_container.go

```diff
@@ -150,10 +158,53 @@ var _ = SIGDescribe("Checkpoint Container [NodeFeature:CheckpointContainer]", fu
 		}
 
 		framework.ExpectNoError(err)
-		// TODO: once a container engine implements the Checkpoint CRI API this needs
-		// to be extended to handle it.
-		//
+
 		// Checkpointing actually worked. Verify that the checkpoint exists and that
 		// it is a checkpoint.
+
+		raw, err := result.Raw()
+		framework.ExpectNoError(err)
+		answer := checkpointResult{}
+		err = json.Unmarshal(raw, &answer)
+		framework.ExpectNoError(err)
+
+		for _, item := range answer.Items {
+			// Check that the file exists
+			_, err := os.Stat(item)
+			framework.ExpectNoError(err)
+			// Check the content of the tar file
+			// At least looking for the following files
+			//  * spec.dump
+			//  * config.dump
+			//  * checkpoint/inventory.img
+			// If these files exist in the checkpoint archive it is
+			// probably a complete checkpoint.
+			specDumpExists := false
+			configDumpExists := false
+			inventoryExists := false
```

Comment:
WDYT about having a map[string]bool that maps path to whether it exists? I think it would simplify and make it easier to add cases in the future

Reply:
PTAL at my latest changes. I tried to use a map.